python/gradbench/evaluation.py

```diff
@@ -84,7 +84,10 @@ def send(self, message: Any) -> Any:
         print(flush=True)
         if message["kind"] == "end":
             return
-        response = json.loads(sys.stdin.readline())
+        l = sys.stdin.readline()
+        if l == "":
+            raise EOFError
```

Comment:
Huh, I didn't realize `readline` would return an empty string in this situation. Can we add a unit test for this? I've been thinking for a while that we should start adding unit tests for things like this that don't happen during "normal" execution (so they're not currently being checked in CI).

Reply:
Yeah we don't need to do it in this PR, but it'd be nice to do at some point; I opened  #180 to note this.